MobileBuy/buy/src/main/java/com/shopify/buy/dataprovider/IBuyClient.java

```diff
@@ -0,0 +1,210 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2015 Shopify Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.shopify.buy.dataprovider;
+
+import com.shopify.buy.model.AccountCredentials;
+import com.shopify.buy.model.Address;
+import com.shopify.buy.model.Checkout;
+import com.shopify.buy.model.Collection;
+import com.shopify.buy.model.CreditCard;
+import com.shopify.buy.model.Customer;
+import com.shopify.buy.model.CustomerToken;
+import com.shopify.buy.model.GiftCard;
+import com.shopify.buy.model.Order;
+import com.shopify.buy.model.Product;
+import com.shopify.buy.model.ShippingRate;
+import com.shopify.buy.model.Shop;
+
+import java.util.List;
+
+import rx.Observable;
+import rx.Scheduler;
+
+public interface IBuyClient {
```

Comment:
Would be nice to organize this file a bit more with some sectioning/documentation.
ie. Products, Customer (Orders, Address), and Checkout apis in their own sections.
The interface is quite large and the interleaving of the RxJava endpoints with the other endpoints makes it a bit hard to grok. 
Maybe organized by section, and by return type?